plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnloadConnectorTest.java

```diff
@@ -0,0 +1,161 @@
+        assertThat(splitSource).isInstanceOf(RedshiftUnloadSplitSource.class);
+
+        ConnectorSplitSource.ConnectorSplitBatch connectorSplitBatch = splitSource.getNextBatch(2).get();
+        assertThat(connectorSplitBatch.isNoMoreSplits()).isTrue();
+        assertThat(connectorSplitBatch.getSplits()).hasSize(2);
+        connectorSplitBatch.getSplits().forEach(split -> assertThat(split).isInstanceOf(RedshiftUnloadSplit.class));
+        verifyS3UnloadedFiles(connectorSplitBatch.getSplits().stream().map(split -> ((RedshiftUnloadSplit) split).path()).toList());
+    }
+
+    @Test
+    void testUnloadFlowFallbackToJdbc()
+            throws ExecutionException, InterruptedException
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(TEST_SCHEMA, "nation");
+        JdbcTableHandle table = new JdbcTableHandle(
+                new JdbcNamedRelationHandle(schemaTableName, new RemoteTableName(Optional.of(TEST_DATABASE), Optional.of(TEST_SCHEMA), "nation"), Optional.empty()),
+                TupleDomain.all(),
+                ImmutableList.of(),
+                Optional.empty(),
+                OptionalLong.of(2),
+                Optional.empty(),
+                Optional.of(ImmutableSet.of()),
+                0,
+                Optional.empty(),
+                ImmutableList.of());
+
+        ConnectorSession session = createUnloadSession();
+        try (ConnectorSplitSource splitSource = redshiftUnloadSplitManager.getSplits(null, session, table, DynamicFilter.EMPTY, Constraint.alwaysTrue())) {
+            assertThat(splitSource).isInstanceOf(FixedSplitSource.class);
+
+            ConnectorSplitSource.ConnectorSplitBatch connectorSplitBatch = splitSource.getNextBatch(2).get();
+            assertThat(connectorSplitBatch.isNoMoreSplits()).isTrue();
+            assertThat(connectorSplitBatch.getSplits()).hasSize(1);
+            connectorSplitBatch.getSplits().forEach(split -> assertThat(split).isInstanceOf(JdbcSplit.class));
+        }
+    }
+
+    private static void verifyS3UnloadedFiles(List<String> files)
+    {
+        files.forEach(file -> assertThat(file).startsWith(S3_UNLOAD_ROOT));
+        try (S3Client s3 = S3Client.builder()
+                .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create(UNLOAD_AWS_ACCESS_KEY, UNLOAD_AWS_SECRET_KEY)))
+                .region(Region.US_EAST_1)
+                .build()) {
+            files.forEach(
+                    file -> {
+                        URI s3Path = URI.create(file);
+                        String bucket = s3Path.getHost();
+                        String key = s3Path.getPath().substring(1);
+                        assertThat(s3.headObject(request -> request.bucket(bucket).key(key)).contentLength()).isGreaterThan(0);
+                        s3.deleteObject(request -> request.bucket(bucket).key(key));
```

Comment:
Removing objects in "verify" method looks confusing.

Reply:
added a separate method